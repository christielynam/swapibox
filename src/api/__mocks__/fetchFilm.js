const mockFilm = {
  title: 'Some title',
  opening_crawl: 'Some text',
  release_date: 'Some date'
}

const fetchFilm = jest.fn().mockImplementation(() => Promise.resolve(mockFilm))

export default fetchFilm