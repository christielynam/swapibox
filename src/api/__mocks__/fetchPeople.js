const fetchPeople = jest.fn()

export default fetchPeople