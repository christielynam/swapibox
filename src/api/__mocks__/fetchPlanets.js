const mockPlanets = [
  {
    favorited: false, 
    climate: 'temperate',
    name: 'Alderaan',
    population: '200000000',
    residents: ['Leia', 'Bail', 'Raymus'],
    terrain: 'grasslands, mountains',
    type: 'planets'
  }
]

const fetchPlanets = jest.fn().mockImplementation(() => Promise.resolve(mockPlanets))

export default fetchPlanets