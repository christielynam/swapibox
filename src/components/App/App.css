.App {
  box-sizing: border-box;
  min-height: 100vh;
  padding: 0;
  margin: 0;
  background-color: #081F29;
  text-align: center;
}

.app-title {
  margin: 0;
  padding-top: 25px;
  color: yellow;
  font-size: 50px;
}

img {
  padding-top: 25px;
}

