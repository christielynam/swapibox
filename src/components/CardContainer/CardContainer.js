import React from 'react'
import loading from '../../assets/loading.gif'
import Card from '../Card/Card'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center
`

const CardContainer = ({ data, toggleFavorite }) => {
  const displayCards = data.map(item => (
    <Card item={item} toggleFavorite={toggleFavorite} key={item.name} />
  ))

  return (
    <Container>
      {data.length ? displayCards : <img src={loading} alt='loading gif' />}
    </Container>
  )
}


export default CardContainer