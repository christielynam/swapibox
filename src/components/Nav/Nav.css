.nav-link {
  height: 40px;
  width: 150px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border-radius: 10px;
  font-size: 15px;
  background-color: white;
  color: #081F29;
  text-decoration: none;
}
  
.active {
  background-color: yellow;
}