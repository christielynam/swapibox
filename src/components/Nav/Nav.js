import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const NavBar = styled.div`
  display: flex;
  justify-content: space-evenly;
  padding-top: 25px;
  button {
    height: 40px;
    width: 150px;
    border-radius: 10px;
    font-size: 15px;
  };
`

const Nav = () => {
  return (
    <NavBar>
      <NavLink to='/people'><button>PEOPLE</button></NavLink>
      <NavLink to='/planets'><button>PLANETS</button></NavLink>
      <NavLink to='/vehicles'><button>VEHICLES</button></NavLink>
      <NavLink to='/favorites'><button>FAVORITES [num]</button></NavLink>
    </NavBar>
  )
}

export default Nav;
