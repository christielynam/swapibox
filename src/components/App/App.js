import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Nav from '../Nav/Nav';
import FilmScroll from '../FilmScroll/FilmScroll';
import './App.css';
import { fetchFilm } from '../../utils/apiCalls';
import { cleanFilm } from '../../utils/cleaner';

class App extends Component {
  constructor() {
    super()
    this.state = {
      film: {},
      people: [],
      planets: [],
      vehicles: []
    }
  }

  async componentDidMount () {
    const num = Math.floor(Math.random() * 7 + 1)
    const fullFilm = await fetchFilm(num)
    const film = cleanFilm(fullFilm)
    this.setState({film})
  }

  filmLoading = () = {
    
  }
  
  render() {
    return (
      <div className="App">
        <h1 className='app-title'>SwapiBox</h1>
        <Route path='/' component={Nav} />
        <Route exact path='/' render={() => (
          <FilmScroll film={this.state.film} />)} 
        />
      </div>
    );
  }
}

export default App;
