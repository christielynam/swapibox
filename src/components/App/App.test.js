import React from 'react';
import App from './App';
import { MemoryRouter } from 'react-router'
import { shallow, mount } from 'enzyme'
import Nav from '../Nav/Nav'
import CardContainer from '../CardContainer/CardContainer';
import fetchFilm from '../../api/fetchFilm'
import fetchPeople from '../../api/fetchPeople'
import fetchPlanets from '../../api/fetchPlanets'
import fetchVehicles from '../../api/fetchVehicles'

jest.mock('../../api/fetchFilm.js')
jest.mock('../../api/fetchPeople.js')
jest.mock('../../api/fetchPlanets.js')
jest.mock('../../api/fetchVehicles.js')

describe('App', () => {
  let wrapper
  
  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  afterEach(() => {
    fetchPeople.mockClear()
    fetchPlanets.mockClear()
    fetchVehicles.mockClear()
  })

  it('matches the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    wrapper = shallow(<App />, { disableLifecycleMethods: true })
    const defaultState = {
      film: {},
      people: [],
      planets: [],
      vehicles: []
    }

    expect(wrapper.state()).toEqual(defaultState)
  })

  it('CDM calls fetchFilm', async () => {

    expect(fetchFilm).toHaveBeenCalled()
  })

  it('updates state with a random film', async () => {
    const expected = {
      title: 'Some title',
      opening_crawl: 'Some text',
      release_date: 'Some date'
    }
    await fetchFilm()

    expect(wrapper.state('film')).toEqual(expected)
  })

  it('getPeople calls fetchPeople', () => {
    wrapper.instance().getPeople()

    expect(fetchPeople).toHaveBeenCalled()
  })

  it('getPeople updates state with an array of people', async () => {
    const expected = [
      {
        favorited: false, 
        homeworld: 'Tatooine',
        name: 'Luke Skywalker',
        population: '200000',
        species: 'Human',
        type: 'people'
      }
    ]
    await wrapper.instance().getPeople()

    expect(wrapper.state('people')).toEqual(expected)
  })

  it('should not call fetchPeople if there are already people in state',async () => {
    wrapper.setState({
      people: [{
        favorited: false, 
        homeworld: 'Tatooine',
        name: 'Luke Skywalker',
        population: '200000',
        species: 'Human',
        type: 'people'
      }]
    })

    await wrapper.instance().getPeople()

    expect(fetchPeople).not.toHaveBeenCalled()
  })


  it('getPlanets calls fetchPlanets', () => {
    wrapper.instance().getPlanets()

    expect(fetchPlanets).toHaveBeenCalled()
  })

  it('getPlanets updates state with an array of planets', async () => {
    const expected = [
      {
        favorited: false, 
        climate: 'temperate',
        name: 'Alderaan',
        population: '200000000',
        residents: ['Leia', 'Bail', 'Raymus'],
        terrain: 'grasslands, mountains',
        type: 'planets'
      }
    ]
    await wrapper.instance().getPlanets()

    expect(wrapper.state('planets')).toEqual(expected)
  })

  it('should not call fetchPlanets if there are already planents in state', async () => {
    wrapper.setState({
      planets: [{
        favorited: false, 
        climate: 'temperate',
        name: 'Alderaan',
        population: '200000000',
        residents: ['Leia', 'Bail', 'Raymus'],
        terrain: 'grasslands, mountains',
        type: 'planets'
      }]
    })

    await wrapper.instance().getPlanets()

    expect(fetchPlanets).not.toHaveBeenCalled()
  })

  it('getVehicles calls fetchVehicles', () => {
    wrapper.instance().getVehicles()

    expect(fetchVehicles).toHaveBeenCalled()
  })

  it('getVehicles updates state with an array of vehicles', async () => {
    const expected = [
      {
        name: 'Sand Crawler',
        model: 'Digger Crawler',
        class: 'wheeled',
        passengers: '30',
        type: 'vehicles',
        favorited: false
      }
    ]
    await wrapper.instance().getVehicles()

    expect(wrapper.state('vehicles')).toEqual(expected)
  })

  it('should not call fetchVehicles if there are already vehicles in state', async () => {
    wrapper.setState({
      vehicles: [{
        name: 'Sand Crawler',
        model: 'Digger Crawler',
        class: 'wheeled',
        passengers: '30',
        type: 'vehicles',
        favorited: false
      }]
    })

    await wrapper.instance().getVehicles()

    expect(fetchVehicles).not.toHaveBeenCalled()
  })

  it('toggleFavorite toggles favorite from false to true on an item', () => {
    const mockPeople = [{name: 'Christie', homeworld: 'Earth', population: '1', species: 'Human', type: 'people', favorited: false}, {name: 'Will', homeworld: 'Earth', population: '100', species: 'Human', type: 'people', favorited: false}]

    wrapper.setState({ people: mockPeople })

    const expected = [{name: 'Christie', homeworld: 'Earth', population: '1', species: 'Human', type: 'people', favorited: true}, {name: 'Will', homeworld: 'Earth', population: '100', species: 'Human', type: 'people', favorited: false}]

    wrapper.instance().toggleFavorite('people', 'Christie')

    expect(wrapper.state('people')).toEqual(expected)
  })

  it('toggleFavorite toggles favorite from true to false on an item', () => {
    const mockPeople = [{name: 'Christie', homeworld: 'Earth', population: '1', species: 'Human', type: 'people', favorited: true}, {name: 'Will', homeworld: 'Earth', population: '100', species: 'Human', type: 'people', favorited: false}]

    wrapper.setState({ people: mockPeople })

    const expected = [{name: 'Christie', homeworld: 'Earth', population: '1', species: 'Human', type: 'people', favorited: false}, {name: 'Will', homeworld: 'Earth', population: '100', species: 'Human', type: 'people', favorited: false}]

    wrapper.instance().toggleFavorite('people', 'Christie')

    expect(wrapper.state('people')).toEqual(expected)
  })

  it('collectFavorites returns an array of all the favorited items', () => {
    const mockPeople = [{name: 'Christie', homeworld: 'Earth', population: '1', species: 'Human', type: 'people', favorited: true}, {name: 'Will', homeworld: 'Earth', population: '100', species: 'Human', type: 'people', favorited: false}]
    const mockPlanets = [{favorited: false, climate: 'temperate', name: 'Alderaan', population: '200000000', residents: ['Leia', 'Bail', 'Raymus'], terrain: 'grasslands, mountains', type: 'planets'}, {favorited: true, climate: 'humid', name: 'Alabama', population: '100000', residents: ['Christie', 'Hudson'], terrain: 'rolling hills', type: 'palnets'}]
    const mockVehicles = [{name: 'Sand Crawler', model: 'Digger Crawler', class: 'wheeled', passengers: '30', type: 'vehicles', favorited: false}, {name: 'Range Rover', model: 'Sport HSE', class: 'AWD', passengers: '5', type: 'vehicles', favorited: true}]

    wrapper.setState({
      people: mockPeople,
      planets: mockPlanets,
      vehicles: mockVehicles
    })
    
    const expected = [{name: 'Christie', homeworld: 'Earth', population: '1', species: 'Human', type: 'people', favorited: true}, {favorited: true, climate: 'humid', name: 'Alabama', population: '100000', residents: ['Christie', 'Hudson'], terrain: 'rolling hills', type: 'palnets'}, {name: 'Range Rover', model: 'Sport HSE', class: 'AWD', passengers: '5', type: 'vehicles', favorited: true}]

    expect(wrapper.instance().collectFavorites()).toEqual(expected)
  })

})

describe('Routes', () => {
  it('should render the Nav component when at the root route', () => {
    const wrapper = mount(<MemoryRouter initialEntries={['/']}>
      <App />
    </MemoryRouter>)

    expect(wrapper.find(Nav)).toHaveLength(1)
  })

  it('should render the CardContainer when at the people route', () => {
    const wrapper = mount(<MemoryRouter initialEntries={['/people']}>
      <App />
    </MemoryRouter>)

    expect(wrapper.find(CardContainer)).toHaveLength(1)
  })

  it('should render the CardContainer when at the planets route', () => {
    const wrapper = mount(<MemoryRouter initialEntries={['/planets']}>
      <App />
    </MemoryRouter>)

    expect(wrapper.find(CardContainer)).toHaveLength(1)
  })

  it('should render the CardContainer when at the vehicles route', () => {
    const wrapper = mount(<MemoryRouter initialEntries={['/vehicles']}>
      <App />
    </MemoryRouter>)

    expect(wrapper.find(CardContainer)).toHaveLength(1)
  })

  it('should render the CardContainer when at the favorites route', () => {
    const wrapper = mount(<MemoryRouter initialEntries={['/favorites']}>
      <App />
    </MemoryRouter>)

    expect(wrapper.find(CardContainer)).toHaveLength(1)
  })
})



  
  