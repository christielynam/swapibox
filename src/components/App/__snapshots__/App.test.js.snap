// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="app-title"
        >
          SwapiBox
        </h1>,
        <Route
          component={[Function]}
          path="/"
        />,
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SwapiBox",
          "className": "app-title",
        },
        "ref": null,
        "rendered": "SwapiBox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="app-title"
          >
            SwapiBox
          </h1>,
          <Route
            component={[Function]}
            path="/"
          />,
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SwapiBox",
            "className": "app-title",
          },
          "ref": null,
          "rendered": "SwapiBox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
